common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java

```diff
@@ -46,6 +53,10 @@
 public abstract class BlockStoreClient implements Closeable {
   protected final SparkLogger logger = SparkLoggerFactory.getLogger(this.getClass());
 
+  private static final ScheduledExecutorService scheduledExecutor =
+          Executors.newScheduledThreadPool(16,
+                  NettyUtils.createThreadFactory("Block Store Client Retry"));
```

Comment:
Why do we need `16` threads here ?
Other than creation of a `TransportCLient`, there is nothing else blocking here right ?

Was that the intent for higher thread count ?

Reply:
Now use `Executors.newSingleThreadScheduledExecutor`